markets/spot-market/contracts/modules/WrapperModule.sol

```diff
@@ -110,6 +111,10 @@ contract WrapperModule is IWrapperModule {
             SpotMarketFactory.TransactionType.WRAP
         );
 
+        if (amountToMint < minAmountReceived) {
+            revert InsufficientAmountReceived(minAmountReceived, amountToMint);
+        }
+
```

Comment:
lets move this and the line above that initializes `amountToMint` before the fee collection logic, so we can revert prior to any other calculations/withdrawing from mm